src/map.cpp

```diff
@@ -6171,12 +6173,38 @@ void map::save()
     }
 }
 
+void map::initialize_map_extras()
+{
+    tinymap *self = dynamic_cast<tinymap *>( this );
```

Comment:
Why the dynamic cast and not a simple virtual function (overridden in the `tinymap` to do nothing)? And why are you storing the returned pointer at all? All you do is compare it to `nullptr`, this can be done directly (e.g. `if( !dynamic_cast<...>(...) )`).

Reply:
Not sure what you meant by making a virtual function overridden in `tinymap`?  Changed it to the more direct way for now though.